### PR TITLE
Bootupcheck and import/export enhancements

### PR DESCRIPTION
**Pull request by iv-gha**
Hello Gareth,

I thought you might like to consider some import/export tweaks - these aim at improving the exchange of songs with others (with less 'missing' info and less manual correction following import!).  See if you think the changes are useful.  Your knowledge of the formats is stronger than mine.

Bootupcheck tweaks - also for consideration.

Please see the commit notes.

Thanks again for openSongApp!

Kind regards
Ian

### CODE REVIEW

File: app/src/main/java/com/garethevans/church/opensongtablet/ExportPreparer.java [hunk 1]
```diff
@@ -824,25 +850,31 @@ private String prepareOnSongFile(Context c, ProcessSong processSong) {
...
+        s.append("Tempo: ").append(StaticVariables.mTempo).append("\n");
+        s.append("Time: ").append(StaticVariables.mTimeSig).append("\n");
         // Go through each song section and add the ChordPro formatted chords
         for (int f = 0; f< StaticVariables.songSections.length; f++) {
             // IV - Quick exit if Heading or Note
             if (!StaticVariables.songSections[f].startsWith(" B_") && !StaticVariables.songSections[f].startsWith(";__")) {
                 s.append(processSong.songSectionChordPro(c, f, true));
             }
         }
+        s.append("CCLI: ").append(StaticVariables.mCCLI).append("\n");
```
Comment (thebigg73) [inline app/src/main/java/com/garethevans/church/opensongtablet/ExportPreparer.java:863]: These have to be above the main content - headers in the documents

**Review by thebigg73 — COMMENTED** [review]
Only issue on initial inspection is the export prepare for OnSong/ChoPro - the copyright/ccli can't go at the bottom of the file (as footers).  The formats expect all headers at the top.  I'll adjust

**Review by thebigg73 — APPROVED** [review]

**Comment by iv-gha** [comment]
OK.

I found that ccli format does put ccli and copyright at the bottom. I guess
opensong have to cope.  My gut feeling is that chordPro format should not
have an order expectation - I can look.

Ccli files also have 'noise' at the bottom which I try to skip - as you
will see in the code.

I think the txt format should have copyright at the bottom as that is
really a print format.

Thanks for considering.

Kind regards
Ian

ADDED : ChordPro spec is not too clear on directives positions.  However v6 is out with  'start_of_...' directives for bridge etc.  At some point openSongApp may have to support v6 features.


End of a CCLI chordPro file!

{comment: Ending}
Jesus [Ab2]Christ, my [Bbsus]liv -  [Bb]ing   [Cm]hope    [Eb/G]
Oh [Eb/Bb]God,    [Ab2]You are my [Eb/Bb]liv -  [Bb]ing   [Eb]hope
{ccli_license: 2057}
{ccli: 7106807}
{copyright: 2017 Phil Wickham Music, Simply Global Songs, Sing My Songs, Bethel Music Publishing}

**Comment by thebigg73** [comment]
Hi Ian,

Yeah, I had a quick read on the directives and while most (with the exception of contents and start/end of ones) tended to be at the top, it doesn't state that they need to be.  I guess I just assumed they were at the top of the file. I've put the order back to having them at the bottom as you had.
V6 changes will need to come at some point!

The variations issue (export) will be sorted in the new material app version as it deals with songs as objects (not static variables) and will be easier to include the export of all files (pdf, variations, images, custom slides, notes, etc.) without adding them to the database.  I didn't want variations in the database as they are temporary files that are exported/imported as part of sets.  That's was so they didn't overwrite user's versions of songs and clog up the storage space.  They get created when a set containing a variation (a custom slides that is embedded in the set file) is loaded by the app and that folder is cleared when a new set is loaded in.  It would require a fairly long winded rehash of code to load the XML, parse it into new variables and create an expotable PDF (without overwriting the currently doing loaded into the static variables mTitle, mLyrics, etc.).  I think it's sensible to live with this for this version given it will be ready to implement in the rewrite I'm working on.

**Comment by thebigg73** [comment]
I'll do some more testing today of your changes

**Pull request merged**